### PR TITLE
Lesson 5 Assignment: Added exception handling for prompt and donation amount

### PR DESCRIPTION
**Pull request by ajayrandhawa1**

### CODE REVIEW

File: students/Ajay_Randhawa/lesson05/mailroom_Part_3.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    print("\n")
+    print(string_header)
+    print("-----------------------------------------------------------")
+    #printing the values in the donorlist
+    for donor, (total, number, average) in sorted_list:
+        string = '{:20}{:>2}{:10}{:11}{:>5}{:11}'.format(donor, symbol, total, number, symbol, average)
+        print(string)
+        print("\n")
+
+def lettersToAllDonors():
```
Comment (charlesrobison) [inline students/Ajay_Randhawa/lesson05/mailroom_Part_3.py:88]: Function should be snake_case, just like your other functions

File: students/Ajay_Randhawa/lesson05/mailroom_Part_3.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    for donor, (total, number, average) in donorlist.items():
+        letter = "Dear %s,\n\n    Thank you for your very kind donation of %s.\n    It will be put to very good use.\n\nSincerely, \n-The Team"%(donor, total)
+
+        filename = donor.replace(" ", "_") + ".txt"
+        with open(filename, 'w') as g:
+            g.write(str(letter))
```
Comment (charlesrobison) [inline students/Ajay_Randhawa/lesson05/mailroom_Part_3.py:94]: Sorting isn't working correctly - report should be sorted by total given in descending order

File: students/Ajay_Randhawa/lesson05/mailroom_Part_3.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
+#Lesson 5 assignment
+import sys
+from operator import itemgetter
```
Comment (charlesrobison) [inline students/Ajay_Randhawa/lesson05/mailroom_Part_3.py:3]: itemgetter imported but not used

File: students/Ajay_Randhawa/lesson06/mailroom_Part_4.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
+#Lesson 5 assignment
+import sys
+from operator import itemgetter
```
Comment (charlesrobison) [inline students/Ajay_Randhawa/lesson06/mailroom_Part_4.py:3]: itemgetter imported but not used

File: students/Ajay_Randhawa/lesson06/mailroom_Part_4.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    print("\n")
+    print(string_header)
+    print("-----------------------------------------------------------")
+    #printing the values in the donorlist
+    for donor, (total, number, average) in sorted_list:
+        string = '{:20}{:>2}{:10}{:11}{:>5}{:11}'.format(donor, symbol, total, number, symbol, average)
+        print(string)
+        print("\n")
+
+def lettersToAllDonors():
```
Comment (charlesrobison) [inline students/Ajay_Randhawa/lesson06/mailroom_Part_4.py:105]: Should be snake_case

File: students/Ajay_Randhawa/lesson06/mailroom_Part_4.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    for donor, (total, number, average) in donorlist.items():
+        filename = donor.replace(" ", "_") + ".txt"
+        with open(filename, 'w') as g:
+            g.write(str(get_letter_text(donor, total)))
```
Comment (charlesrobison) [inline students/Ajay_Randhawa/lesson06/mailroom_Part_4.py:109]: Please fix sorting on this version as well

**Review by charlesrobison — APPROVED** [review]
Couple of suggestions and both versions of mailroom need to have the sorting corrected.  Tests look good and do pass.  Please submit your code on the course site for grading.  Let me know if you have questions.

Ready to merge

File: students/Ajay_Randhawa/lesson06/mailroom_Part_4.py (hunk 6 above)
Comment (ajayrandhawa1) [reply students/Ajay_Randhawa/lesson06/mailroom_Part_4.py:109]: I am not sure if/how sorting is needed here?

Comment (charlesrobison) [reply students/Ajay_Randhawa/lesson06/mailroom_Part_4.py:113]: I made the comment in the wrong place - should be in the create_report() function - same as prior version.  Report should be sorted by total given in descending order.

**Review by ajayrandhawa1 — COMMENTED** [review]
Lesson5 (Assignment#4) feedback incorporated

**Review by charlesrobison — APPROVED** [review]
@ajayrandhawa1 please submit your mailroom files to a new PR if you would like another review.

**Pull request merged**